scanamo/src/main/scala/org/scanamo/ops/ScanamoOpsA.scala

```diff
@@ -1,12 +1,14 @@
 package org.scanamo.ops
 
 import com.amazonaws.services.dynamodbv2.model._
+import org.scanamo.ops.ScanamoOpsA.ScanamoResult
 import org.scanamo.request._
 
 sealed trait ScanamoOpsA[A] extends Product with Serializable
-final case class Put(req: ScanamoPutRequest) extends ScanamoOpsA[PutItemResult]
-final case class ConditionalPut(req: ScanamoPutRequest)
-    extends ScanamoOpsA[Either[ConditionalCheckFailedException, PutItemResult]]
+object ScanamoOpsA {
+  type ScanamoResult[A] = Either[AmazonDynamoDBException, A]
```

Comment:
I'm not so sure about this type anymore. Right now Scanamo is a glorified `A => F[B]` and so the way to surface errors is either:
- `A => F[Either[E, B]]`
- `A => F[E, B]`
- `A => F[B]` (i.e. bake error in the semantics of `F`)

The latter option, which we use at the moment, is the easy one. With the caveat that it throws exceptions when using the synchronous client. I think your solution (the first choice, which was already used for conditional puts) is a good step forward because we want to advertise the kind of errors that can happen during an effect. In the future though, I'm inclined to say the second option is the best, because error-handling is usually baked into effect monads and it will be optimised for that (e.g. in zio there is first-class support for fully typed errors, whereas the current cats IO will have to become `EitherT[IO, ?]`).

Reply:
@regiskuckaertz I have gone with the first option for now but it could move towards the second one in the future.